src-web/components/modals/AnsibleAutomationModal.js

```diff
@@ -164,18 +173,64 @@ export class AnsibleAutomationModal extends React.Component {
     })
   }
 
-  handleSubmitClick() {
-    this.modifyPolicyAutomation()
+  async handleSubmitClick() {
+    const {latestJSON, action} = await this.generateJSON()
+    if (latestJSON && action) {
+      const {data:resData} = await this.props.handleModifyPolicyAutomation(latestJSON, action)
+      const errors = _.get(resData, 'modifyPolicyAutomation.errors')
+      if (Array.isArray(errors) && errors.length > 0)  {
+        const error = errors[0]
+        if (_.get(error, 'kind') === 'PolicyAutomation' && _.get(error, 'message')) {
+          this.setState({
+            queryMsg: {
+              msg: _.get(error, 'message'),
+              type: 'danger',
+            }
+          })
+        }
+      } else {
+        this.handleCloseClick('directlyClose')
+      }
+    }
   }
 
-  handleCloseClick() {
+  async handleCloseClick(directlyClose) {
     const { type:modalType, handleClose } = this.props
-    handleClose(modalType)
+    if (directlyClose === 'directlyClose') {
+      handleClose(modalType)
+    } else {
+      const { locale } = this.context
+      const { initialJSON, confirmClose } = this.state
+      const { latestJSON } = await this.generateJSON()
+      let ifChanged = false
+      console.log(JSON.stringify(initialJSON))
+      console.log(JSON.stringify(latestJSON))
+      if (initialJSON && latestJSON) {
+        if (stringify(initialJSON) !== stringify(latestJSON)) {
```

Comment:
can you try using lodash isEqual to compare the object directly?
https://lodash.com/docs/4.17.15#isEqual